### PR TITLE
sw-5440: Edit client test fix

### PR DESCRIPTION
**Pull request by denialanderror**
Aliases the form container, allowing Cypress to handle retries until the form is reattached

### CODE REVIEW

File: cypress/integration/supervision/clients/edit-client.spec.js [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
-//     }
-//   );
-// });
+describe('Edit a client', { tags: ['@supervision', 'client', '@smoke-journey'] }, () => {
+  it(
+    'Given I\'m a Case Manager on Supervision and on the client dashboard page' +
+    'Then the Client Dashboard page loads as expected',
+    () => {
+      cy.get('@clientId').then(clientId => {
+        cy.visit(`/supervision/#/clients/${clientId}/edit`);
```
Comment (mattmachell) [inline cypress/integration/supervision/clients/edit-client.spec.js:12]: Yay for string templating.

File: cypress/integration/supervision/clients/edit-client.spec.js [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
+      const suffix = Math.floor(Math.random() * 10000);
+
+      cy.get('#editClientFormContent').as('edit-panel');
+      cy.get('@edit-panel').within(() => {
+        cy.get('input[name="firstName"]').clear().type('Bill' + suffix);
+        cy.get('input[name="lastName"]').clear().type('Billson' + suffix);
+        cy.get('input[name="memorablePhrase"]').clear().type('Memorable' + suffix);
+      });
+
+      cy.contains('Save & Exit').click();
```
Comment (mattmachell) [inline cypress/integration/supervision/clients/edit-client.spec.js:25]: Is the form save the only button in the DOM with this text at this time?

File: cypress/integration/supervision/clients/edit-client.spec.js [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
+
+      cy.contains('Client details');
+      cy.contains('Bill' + suffix);
```
Comment (mattmachell) [inline cypress/integration/supervision/clients/edit-client.spec.js:28]: Might need to check which element this text is in as the previous form content changes might also trigger it if they haven't gone from the DOM quick enough.

File: cypress/integration/supervision/letters/create-letter.spec.js [hunk 4]
```diff
@@ -44,33 +36,43 @@ describe('Creating, editing and retrieving a letter', { tags: ['@supervision', '
...
         });
       });
 
-      cy.get('.select-template', {timeout: 30000}).contains('Select a template');
+      cy.contains('.select-template', 'Select a template')
       cy.contains('blank: Blank template').click();
 
-      getRecipientHeader().contains('Select recipient/s');
-      getFirstSelectableRecipient().click();
+      cy.get('.smart__action > ng-component:nth-child(2) > div:nth-child(1) > draft-select-recipients:nth-child(1) > div:nth-child(1) > div:nth-child(1) > h2:nth-child(1)')
```
Comment (mattmachell) [inline cypress/integration/supervision/letters/create-letter.spec.js:50]: This is a pretty big selector. Is there any nicer way to target the recipient header element?

File: cypress/integration/supervision/clients/edit-client.spec.js (hunk 2 above)
Comment (denialanderror) [reply cypress/integration/supervision/clients/edit-client.spec.js]: It is. Would prefer it to be addressed differently but that would require Sirius changes.

File: cypress/integration/supervision/clients/edit-client.spec.js (hunk 3 above)
Comment (denialanderror) [reply cypress/integration/supervision/clients/edit-client.spec.js]: I've made this more specific to the table it is looking at.

File: cypress/integration/supervision/clients/edit-client.spec.js (hunk 2 above)
Comment (mattmachell) [reply cypress/integration/supervision/clients/edit-client.spec.js]: Can leave this for now then, but given we might have to live with angular a bit longer worth ticketing up some HTML test hook improvements like adding `data-action="client.save"` style hooks.

**Review by mattmachell — APPROVED** [review]

**Pull request merged**